http/client.go

```diff
@@ -218,6 +207,16 @@ func (c *Client) BinaryRequest(method, url, token string, reqData *RequestData,
 	if reqData.RespReader != nil {
 		reqData.RespReader = resp.Body
 	} else {
+		if method != "HEAD" && resp.ContentLength != 0 {
+			// Read a small amount of data from the response
+			// body so that the client connection can
+			// be reused.
+			size := resp.ContentLength
+			if size > 1024 || size < 0 {
+				size = 1024
+			}
+			resp.Body.Read(make([]byte, size))
```

Comment:
Yes, but I don't think it's worth reading large response bodies to save a connection. In the particular case we're seeing in swift, the body is only about 100 bytes. We could tweak this value later if we find it's not effective.